users/jdelkins/jdelkins.c

```diff
@@ -0,0 +1,274 @@
+            return DOUBLE_HOLD;
+        else
+            return DOUBLE_TAP;
+    } else if (state->count == 3) {
+        if (!state->pressed)
+            return TRIPLE_TAP;
+        else
+            return TRIPLE_HOLD;
+    } else
+        return 8;  // magic number. At some point this method will expand to work for more presses
+}
+
+#endif // TAP_DANCE_ENABLE
+
+__attribute__ ((weak))
+void keyboard_post_init_keymap(void) {
+}
+
+void keyboard_post_init_user(void) {
+    user_config.raw = eeconfig_read_user();
+    keyboard_post_init_keymap();
+}
+
+void eeconfig_init_user(void) {
+    user_config.raw = 0;
+    user_config.system_mac = false;
+    eeconfig_update_user(user_config.raw);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    static uint32_t boot_timer;
+
+    if (!process_record_keymap(keycode, record)) {
+        return false;
+    }
+
+    switch (keycode) {
+        case FW_WRD:
+            do_mac_key(LCTL(KC_RIGHT), ROPT(KC_RIGHT), record);
+            break;
+
+        case BK_WRD:
+            do_mac_key(LCTL(KC_LEFT), ROPT(KC_LEFT), record);
+            break;
+
+        case KB_BOL:
+            do_mac_key(KC_HOME, RCMD(KC_LEFT), record);
+            break;
+
+        case KB_EOL:
+            do_mac_key(KC_END, RCMD(KC_RIGHT), record);
+            break;
+
+        case TG_SYS:
+            if (record->event.pressed) {
+                user_config.system_mac ^= 1;
+                eeconfig_update_user(user_config.raw);
+            }
+            break;
+
+        case KB_COPY:
+            do_mac_key(LCTL(KC_INS), RCMD(KC_C), record);
+            break;
+
+        case KB_PASTE:
+            do_mac_key(LSFT(KC_INS), RCMD(KC_V), record);
+            break;
+
+        case MY_GUI:
+            do_mac_key(KC_LGUI, KC_LOPT, record);
+            break;
+
+        case MY_ALT:
+            do_mac_key(KC_LALT, KC_LCMD, record);
+            break;
+
+        case MY_RGUI:
+            do_mac_key(KC_RGUI, KC_ROPT, record);
+            break;
+
+        case MY_RALT:
+            do_mac_key(KC_RALT, KC_RCMD, record);
+            break;
+
+        case MY_CALC:
+            do_mac_key(KC_CALC, KC_F14, record);
+            break;
+
+        case KB_MAKE:
+            if (!get_mods()) {
+                if (!record->event.pressed)
+                    SEND_STRING("make " QMK_KEYBOARD ":" QMK_KEYMAP SS_TAP(X_ENTER));
+                return false;
+            }
+            break;
+
+        case KB_VRSN:
+            if (!get_mods()) {
+                if (!record->event.pressed) {
+                    if (user_config.system_mac) {
+                        SEND_STRING(QMK_KEYBOARD "/" QMK_KEYMAP " @ " QMK_VERSION " (mac mode)");
+                    } else {
+                        SEND_STRING(QMK_KEYBOARD "/" QMK_KEYMAP " @ " QMK_VERSION " (non-mac mode)");
+                    }
+                }
+                return false;
+            }
+            break;
+
+        case KB_BOOT:
```

Comment:
You can use `RESET` here instead and replace the normal functionality of the keycode.